### ISSUE TITLE
Allow LogicalColumnReader to be IEnumerable<T>

### ISSUE THREAD

**Issue by GPSnoopy** [G-Research/ParquetSharp#16]
Feature requested by user.

Currently they have to write it themselves. Not sure if we want to provide this or not (i.e. whether it's going beyond the scope of this library).

**Comment by GPSnoopy** [comment]
I'd favour a method that returns an IEnumerable from a LogicalColumnReader rather than making the reader IEnumerable.

**Comment by GPSnoopy** [comment]
```csharp
foreach (var value in rowGroupReader.Column(c).LogicalReader().Enumerate())
{
    /* ... */
}
```

**Comment by herebebeasties** [comment]
I'd probably name that `AsEnumerable()` as opposed to `Enumerate()` to be more idiomatic.

But TBH I'm not sure why you'd bother returning a separate `IEnumerable<T>` like that, instead of just making `LogicalColumnReader` implement it. What's the reasoning there? You're already going to need to make an `IEnumerator<T>` which does the right thing, so why would you favour adding another layer?

**Comment by GPSnoopy** [comment]
This should be fixed in 1.4.0.2-beta3. @herebebeasties argument swayed me in favour of implementing `IEnumerable<T>` directly.
